File: FilmesAPI.FSharp/Models/Filme.fs
namespace FilmesAPI.FSharp.Models

open System.ComponentModel.DataAnnotations

[<CLIMutable>]
type Filme = {
    [<Required>] Titulo: string
    [<Required>] Genero: string
    [<StringLength(30)>] Diretor: string
    [<Range(1, 600)>] Duracao: int
    Id: int
}

File: FilmesAPI.Minimal/Models/Filme.cs
using System.ComponentModel.DataAnnotations;

namespace FilmesAPI.Minimal.Models;

public class Filme {
    [Key]
    [Required]
    public int Id { get; set; }

    [Required]
    public string Titulo { get; set; } = string.Empty;

    [Required]
    public string Genero { get; set; } = string.Empty;

    [StringLength(30)]
    public string Diretor { get; set; } = string.Empty;

    [Required]
    public int Duracao { get; set; }
}

